packages/x-components/src/x-modules/search-box/store/actions/set-status.action.ts

```diff
@@ -0,0 +1,19 @@
+import { SearchBoxXStoreModule } from '../types';
+import { StateMachine } from '../../../../services/state-machine.service';
+import { SearchBoxMachine } from '../../search-box.machine';
+
+const machine = new StateMachine(SearchBoxMachine);
```

Comment:
I am not sure where should be created the machine

Reply:
I think here it is fine. Next to the action that uses it. I would suggest to inline the configuration in the constructor `new StateMachine({...})`. Or do you have any plan to use it elsewhere?